### PR TITLE
UIU-3112 - Implement sort and filter functionality on reading room access MCL

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
https://folio-org.atlassian.net/browse/UIU-3112 - Reading Room Access accordion in User record: Sorting and filtering (view mode)

## Note
**The target branch for this PR is not master**
The feature branch will be merged to master very soon, after this PR is merged into feature branch.

## Screencast
chrome_mceVQlFf9z

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/UserDetailSections/ReadingRoomAccess/constant.js [hunk 1]
```diff
@@ -5,3 +5,10 @@ export const rraColumns = {
   NOTES: 'notes',
   ID: 'id'
 };
+
+export const DEFAULT_SORT_ORDER = rraColumns.ACCESS;
+
+export const SORT_DIRECTION = {
+  ASCENDING: 'asc',
+  DESCENDING: 'desc'
+};
```
Comment (Dmytro-Melnyshyn) [inline src/components/UserDetailSections/ReadingRoomAccess/constant.js:14]: Let's use existing [sort types](https://github.com/folio-org/ui-users/blob/3c5aa4d1450d9e9357762c2b516937bd33bf46be/src/constants.js#L51C14-L51C23) constant.

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.js [hunk 2]
```diff
@@ -22,19 +24,60 @@ const ReadingRoomAccess = (props) => {
...
+    rraColumns.READING_ROOM_NAME,
+    rraColumns.NOTES,
+    rraColumns.ID
+  ];
+  const columnMapping = {
+    [rraColumns.ACCESS]: <FormattedMessage id="ui-users.readingRoom.access" />,
+    [rraColumns.READING_ROOM_NAME]: <FormattedMessage id="ui-users.readingRoom.name" />,
+    [rraColumns.NOTES]: <FormattedMessage id="ui-users.readingRoom.note" />,
+    [rraColumns.ID]: <FormattedMessage id="ui-users.readingRoom.lastUpdated" />,
+  };
```
Comment (zburke) [inline src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.js:43]: If you anticipate these lists remaining linked, you might be better served by 
```
const columnMapping = {...};
const visibleColumns = Object.keys(columnMapping);
```
[The traversal order of keys in an object is stable and well-defined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#description) and matches what you want here: order of property creation.

File: src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.test.js [hunk 3]
```diff
@@ -84,13 +87,40 @@ describe('ReadingRoomAccess', () => {
   it('should filter MCL records "Name" column, based on the string entered in search box', async () => {
     render(<ReadingRoomAccess {...alteredProps} />);
     const inputEl = screen.getByPlaceholderText('ui-users.readingRoom.filter');
-    fireEvent.change(inputEl, { target: { value: 'room1' } });
+    fireEvent.change(inputEl, { target: { value: 'room 1' } });
```
Comment (zburke) [inline src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.test.js:90]: [The first sentence in the documentation for `fireEvent` is an explanation of why you should not use fireEvent](https://testing-library.com/docs/dom-testing-library/api-events). You don't _have_ to change this, but please consider the alternative they recommend, [user-event](https://testing-library.com/docs/user-event/intro) for future work.

File: translations/ui-users/en.json [hunk 4]
```diff
@@ -367,7 +367,7 @@
   "readingRoom.name": "Name",
   "readingRoom.note": "Note",
   "readingRoom.lastUpdated": "Last updated",
-  "readingRoom.filter": "Filter",
+  "readingRoom.filter": "Room name",
   "readingRoom.allowed": "Allowed",
   "readingRoom.notAllowed": "Not allowed",
   "readingRoom.metaSection.lastUpdatedDetails": "{date} by {name}",
```
Comment (zburke) [inline translations/ui-users/en.json:373]: Unrelated to this PR, but `{date} by {name}` should be sth like `{date, date, full} by {name}` or `{date, date, short} by {name}` etc. [dateStyle options](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#datestyle).

**Review by zburke — APPROVED** [review]

**Review by manvendra-s-rathore — APPROVED** [review]

**Pull request merged**